### ISSUE TITLE
Should SafeLife environments be pickleable?

### ISSUE THREAD

**Issue by pde** [PartnershipOnAI/safelife#4]
The [MuZero ray code](https://github.com/werner-duvaud/muzero-general/) seems to want gym environments to be pickleable with the cloudpickle module, but currently one gets:

```
>>> import gym
>>> from safelife import safelife_env as sle
>>> sle.SafeLifeEnv.register()                                                       
>>> y = gym.make("safelife-append-still-v1")                                         
>>> import pickle
>>> pickle.dumps(y)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: can't pickle generator objects
```

**Comment by clwainwright** [comment]
Hmm, a good point. I'll work on refactoring the generator into a class-based iterator.

**Comment by clwainwright** [comment]
@pde This should now be fixed on the `clw-dev` branch along with some changes to make random number generators to act consistently across processes. Let me know if you've any comments.

There are a number of useful changes on that branch at this point, so we should think about doing an update and point release some time soon.

**Comment by clwainwright** [comment]
Fixed with v1.1
